File: app/src/main/java/wmding/example/com/mylivetools/module/search/SearchActivity.java
package wmding.example.com.mylivetools.module.search;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import wmding.example.com.mylivetools.R;
import wmding.example.com.mylivetools.data.ArticlesDataRepository;
import wmding.example.com.mylivetools.data.remote.ArticlesDataRemoteSource;

public class SearchActivity extends AppCompatActivity {

    private SearchFragment mSearchFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);

        if (savedInstanceState != null) {
            mSearchFragment = (SearchFragment) getSupportFragmentManager().getFragment(savedInstanceState, SearchFragment.class.getSimpleName());
        }else {
            mSearchFragment = SearchFragment.newInstance();
        }

        new SearchPresenter(ArticlesDataRepository.getInstance(ArticlesDataRemoteSource.getInstance()), mSearchFragment);


        getSupportFragmentManager().beginTransaction()
                .replace(R.id.view_pager, mSearchFragment, SearchFragment.class.getSimpleName())
                .commit();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        if (mSearchFragment.isAdded()) {
            getSupportFragmentManager().putFragment(outState, SearchFragment.class.getSimpleName(), mSearchFragment);
        }
    }
}

File: app/src/main/java/wmding/example/com/mylivetools/module/search/SearchActivity.kt
package wmding.example.com.mylivetools.module.search

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import wmding.example.com.mylivetools.R

class SearchActivity : AppCompatActivity() {

    private var mSearchFragment: SearchFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search)

        if (savedInstanceState != null) {
            mSearchFragment = supportFragmentManager.getFragment(savedInstanceState, SearchFragment::class.java!!.getSimpleName()) as SearchFragment
        } else {
            mSearchFragment = SearchFragment.newInstance()
        }



        supportFragmentManager.beginTransaction()
                .replace(R.id.view_pager, mSearchFragment, SearchFragment::class.java!!.getSimpleName())
                .commit()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        if (mSearchFragment!!.isAdded) {
            supportFragmentManager.putFragment(outState, SearchFragment::class.java!!.getSimpleName(), mSearchFragment)
        }
    }
}
